app/assets/javascripts/common/admins.js

```diff
@@ -1,97 +1,117 @@
+//
+// elements
+//
 const ACCESS_LIST_INPUT_SELECTOR = "input.access-input"
-const facilityAccessDiv = () => document.getElementById("facility-access")
-const selectAllFacilitiesDiv = () => document.getElementById("select-all-facilities")
-const facilityAccessItemsAccessRatio = () => document.getElementsByClassName("access-ratio")
-const facilityAccessItemsPadding = () => document.getElementsByClassName("access-item__padding")
-const facilityAccessPowerUser = () => document.getElementById("facility-access-power-user")
+const $facilityAccessDiv = () => document.getElementById("facility-access")
+const $selectAllFacilitiesContainer = () => document.getElementById("select-all-facilities")
+const $selectAllFacilitiesInput = () => document.getElementById("select-all-facilities-input")
+const $facilityAccessItemsAccessRatio = () => document.getElementsByClassName("access-ratio")
+const $facilityAccessItemsPadding = () => document.getElementsByClassName("access-item__padding")
+const $facilityAccessPowerUser = () => document.getElementById("facility-access-power-user")
+const $accessLevel = () => document.getElementById("access_level")
+const $page = () => $facilityAccessDiv() && $facilityAccessDiv().attributes.getNamedItem('data-page').value
+
 //
-// loads at page refresh
+// load things upfront
 //
-window.addEventListener("DOMContentLoaded", inviteAdmin);
-window.addEventListener("DOMContentLoaded", editAdmin);
+window.addEventListener("DOMContentLoaded", onPageLoad);
 
-function inviteAdmin() {
+function onPageLoad() {
   accessLevelListener()
-  selectAccessLevels()
-  selectAllListener()
+  accessLevelSelector()
+}
+
+//
+// wait for the render_async hook before loading the access tree related JS
+//
+document.addEventListener('render_async_load', function (_event) {
```

Comment:
1/2:

This is one of the main changes in this file, we're loading the access tree related JS after a callback hook from `render_async_load`, once it has finished loading.